File: Python/1047 Game Time with Minutes.py
ih,im, fh, fm = map( int, input( ).split( ))

hs  = fh - ih
min = fm - im

if ih+im/60 > fh+fm/60:
    hs += 24
if im>fm:
    min += 60
    hs -= 1

print('O JOGO DUROU {} HORA(S) E {} MINUTO(S)'.format(hs, min))

# Wrong answer (20%)

File: 1. Begginers/1047-Game_Time_with_Minutes.cpp
#include <iostream>
using namespace std;

int a, b, c, d, hours, minutes;

main(){
    cin >> a>> b>> c>> d;
    hours = c - a;
    minutes = d - b;
    if (hours <= 0){
        hours += 24;
    }
    if (minutes < 0){
        minutes += 60;
        hours--;
    }
    if ((hours == 24) and (minutes > 0)){
        hours -= 24;
    }

    cout<< "O JOGO DUROU "<< hours<< " HORA(S) E "<< minutes<< " MINUTO(S)"<< endl;

}
